lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp

```diff
@@ -436,10 +436,11 @@ struct ReduceOpConversion
         inputTy.getEncoding().dyn_cast<triton::gpu::MfmaEncodingAttr>();
       if (inMfma && inMfma.getIsTransposed()) {
         assert(numLaneToReduce == 2 || numLaneToReduce == 4);
-        // for mfma 32x32 adjecant threads in y dimension in transposed MFMA layout are 32
-        // apart: [[0 0 0 0 32 32 32 32 ...] [1 1 1 1 33 33 33 33 ...] ...].
-        // for mfma 16x16 adjecant threads in y dimension in transposed MFMA layout are 16
-        // apart: [[0 0 0 0 16 16 16 16 32 32 32 32 ...] [1 1 1 1 33 33 33 33 ...] ...].
+        // for mfma 32x32 adjacent threads in y dimension in transposed MFMA
+        // layout are 32 apart: [[0 0 0 0 32 32 32 32 ...] [1 1 1 1 33 33 33 33
+        // ...] ...]. for mfma 16x16 adjacent threads in y dimension in
+        // transposed MFMA layout are 16 apart: [[0 0 0 0 16 16 16 16 32 32 32
+        // 32 ...] [1 1 1 1 33 33 33 33 ...] ...].
```

Comment:
Is it possible to get the waveSize from the gpu dialect or mfma layout?

Reply:
Unfortunately no...
However! MFMA layout appears in IR only if target is CDNA architecture, which has only 64 waves mode.

I think it should be safe to use constant here.
In my opinion we should report MFMA layout on non CDNA GPU as an error.

Reply:
If you really want, it is possible to infer waveSize from mfmaLayout by computing a product of mfmaLayout.threadsPerWarp. But that is a little "ugly" in my opinion.

Reply:
And you can also get it from gpu dialect like here: https://github.com/ROCmSoftwarePlatform/triton/blob/4d539d7dae055bb6b8dbb1b2b380118333250f15/lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp#L589